File: .js/hillcypher.js
function arrToString(cypher) {
    let s = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    let res = "";
    for(let i of cypher ){
        res += s[i];
    }
    return res;
}

function keyMatrix(key, n) {
    let res = [];
    let ch = 0;
    for(let i=0; i<n; i++) {
        let temp = [];
        for(let j=0; j<n; j++) {
            temp.push( key.charCodeAt(ch) - 65 );
            ch++;
        }
        // console.log(temp)
        res.push(temp);
    }
    
    return res;
}

function plainArr(plain) {
    let res = [];
    for(let i=0; i<plain.length; i++) {
        res.push( plain.charCodeAt(i) - 65 );
    }
    return res;
}

function encrypt(key, plain, n) {
    let res = [];
    for(let i=0; i<n; i++) {
        let temp = 0;
        for(let j=0; j<n; j++) {
            temp += plain[j]*key[j][i];
        }
        res.push(temp);
    }
    for(let i=0; i<n; i++) {
        res[i] %= 26;
    }
    let ans = arrToString(res);
    return ans;
}

function encrypt2(key, plain, n) {
    let res = [];
    for(let i=0; i<n; i++) {
        let temp = 0;
        for(let j=0; j<n; j++) {
            temp += key[i][j] * plain[j];
        }
        res.push(temp);
    }
    for(let i=0; i<n; i++) {
        res[i] %= 26;
    }
    let ans = arrToString(res);
    return ans;
}





let key = "GYBNQKURP", plain = "ACT";

let keyMat = keyMatrix(key, 3);
let plainA = plainArr(plain);

let cypher1 = encrypt(keyMat, plainA, plain.length );
let cypher2 = encrypt2(keyMat, plainA, plain.length)
console.log("CYPHER TEXT 1 -> ",cypher1);
console.log("CYPHER TEXT 2 -> ",cypher2)

File: .py/hillCypher.py
def arrToString(cypher):
    s = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    res= ""
    for i in cypher:
        res += s[i]
    
    return res
    
def keyMatrix(key, n):
    res = [[0]*n for i in range(n)]
    ch = 0
    for i in range(n):
        for j in range(n):
            res[i][j] = (ord(key[ch]) - 65)
            ch += 1
    return res
 
def plainArr(plain):
    res = []
    for i in plain:
        res.append(ord(i)-65)
    return res

def encrypt(key, plain, n):
    res = []
    for i in range(0,n):
        temp = 0
        for j in range(0,n):
            temp += plain[j]*key[j][i]
        res.append(temp)  
    for i in range(n):
        res[i] %= 26
        
    res = arrToString(res)
    return res
    
def encrypt2(key, plain, n):
    res = []
    
    for i in range(n):
        temp = 0
        for j in range(n):
            temp += key[i][j] * plain[j]
        res.append(temp)
    
    for i in range(n):
        res[i] %= 26
        
    res = arrToString(res)
    return res





key = "GYBNQKURP"
plain = "ACT"

keyMat = keyMatrix(key, 3)
plainA = plainArr(plain)

cypher1 = encrypt(keyMat, plainA, len(plain))
cypher2 = encrypt2(keyMat, plainA, len(plain))
print("CYPHER TEXT 1 -> ",cypher1)
print("CYPHER TEXT 2 -> ",cypher2)













